msdk-io/msdk-io-csv/src/main/java/io/github/msdk/io/csv/CsvFileExportMethod.java

```diff
@@ -0,0 +1,264 @@
+            return null;
+        }
+
+        // Get number of rows
+        totalRows = featureTable.getRows().size();
+
+        // Write data to CSV file
+        writeData(featureTable, writer);
+
+        // Close file
+        try {
+            writer.close();
+        } catch (Exception e) {
+            logger.info("Could not close file " + csvFile);
+            return null;
+        }
+
+        return csvFile;
+    }
+
+    private void writeData(FeatureTable featureTable, FileWriter writer) {
+
+        // Buffer for writing
+        StringBuffer line = new StringBuffer();
+
+        // Write column headers
+        for (FeatureTableColumn<?> column : columns) {
+            String columnName = column.getName();
+            Sample sample = column.getSample();
+
+            // Add sample name to column name if available
+            if (sample != null) {
+                columnName = sample.getName() + " " + columnName;
+            }
+
+            line.append(escapeStringForCSV(columnName) + separator);
+        }
+
+        // Remove last separator
+        String stringLine = removeLastCharacter(line.toString());
+
+        // Write the line to the CSV file
+        try {
+            writer.write(stringLine);
+        } catch (Exception e) {
+            logger.info("Could not write to file " + csvFile);
+            return;
+        }
+
+        if (canceled)
+            return;
+
+        // Write data values
+        List<FeatureTableRow> rows = featureTable.getRows();
+        for (FeatureTableRow row : rows) {
+
+            // Reset the buffer
+            line.setLength(0);
+
+            // Loop through all columns
+            for (FeatureTableColumn<?> column : columns) {
+                Object object = row.getData(column);
+                String strValue = "";
+
+                if (object == null) {
+                    strValue = "";
+                }
+                // Chromatography Info
+                else if (object instanceof ChromatographyInfo) {
+                    ChromatographyInfo chromatographyInfo = (ChromatographyInfo) object;
+                    Float rt1 = chromatographyInfo.getRetentionTime();
+                    Float rt2 = chromatographyInfo.getSecondaryRetentionTime();
+
+                    if (rt1 != null) {
+                        strValue = rt1.toString();
+                    }
+                    if (strValue != null && rt2 != null) {
+                        strValue = strValue + itemSeparator;
+                    }
+                    if (rt2 != null) {
+                        strValue = strValue + rt2.toString();
+                    }
+                }
+                // Ion annotations
+                else if (object instanceof List<?>) {
+                    List<IonAnnotation> ionAnnotations = (List<IonAnnotation>) object;
+                    for (IonAnnotation ionAnnotation : ionAnnotations) {
```

Comment:
I can't find a way to check if the object is a List<IonAnnotation> object. Any suggestions?